xonsh/xoreutils/cat.py

```diff
@@ -0,0 +1,104 @@
+import os
+import sys
+
+from xonsh.xoreutils.util import arg_handler
+
+def _cat_single_file(opts, fname, stdin, out, err, line_count=1):
+    if fname == '-':
+        f = stdin
+    elif os.path.isdir(fname):
+        print("cat: {}: Is a directory.".format(fname), file=err)
+        return True, line_count
+    elif not os.path.exists(fname):
+        print("cat: No such file or directory: {}".format(fname), file=err)
+        return True, line_count
+    else:
+        f = open(fname, 'rb')
+    sep = os.linesep.encode()
```

Comment:
I think all of these encode and decode calls should use `$XONSH_ENCODING` and `$XONSH_ENCODING_ERRORS`